src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -469,11 +469,9 @@ protected void setupChild(Configuration childConf)
         try {
             child = moduleFactory.createModule(name);
 
-            if (child instanceof AbstractAutomaticBean) {
-                final AbstractAutomaticBean bean = (AbstractAutomaticBean) child;
-                bean.contextualize(childContext);
-                bean.configure(childConf);
-            }
+            final AbstractAutomaticBean bean = (AbstractAutomaticBean) child;
+            bean.contextualize(childContext);
+            bean.configure(childConf);
```

Comment:
removal of this IF can mean breaking changes, as some thirdparty modules might stop to work.
If module does not need properties or any Checkers/Treewalker properties, it is ok to be not AbstractAutomaticBean, even we do not recommend it and highly expect.

lets keep it as it was before, with IF, to not deal with breaking old and not not well done modules. At least we need to consider to make it as separate and very explicit breaking change.

Reply:
I don't remember details now, but I assume this was required by pitest.

Reply:
lets restore this block and add pitest to suppression for now to clearly see it. As we agree on PR , covering pitest will be final polishing.